plugins/github/app/decorators/stage_decorator.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+Stage.class_eval do
+  validate :github_pull_request_comment_variables
+
+  private
+
+  def github_pull_request_comment_variables
+    if comment = github_pull_request_comment.presence
+      comment % {stage_name: '', reference: ''} # Validate no extra keys are supplied
```

Comment:
call GithubNotification#body instead if duplicating the logic ? ... or use a constant for the supported keys or something like that ...